src/amo/components/AddonCompatibilityError/index.js

```diff
@@ -0,0 +1,93 @@
+/* eslint-disable react/no-danger */
+import { oneLine } from 'common-tags';
+import React, { PropTypes } from 'react';
+import { compose } from 'redux';
+import { connect } from 'react-redux';
+
+import {
+  INCOMPATIBLE_FIREFOX_FOR_IOS,
+  INCOMPATIBLE_NOT_FIREFOX,
+  INCOMPATIBLE_OVER_MAX_VERSION,
+  INCOMPATIBLE_UNDER_MIN_VERSION,
+} from 'core/constants';
+import translate from 'core/i18n/translate';
+import _log from 'core/logger';
+import { sanitizeHTML } from 'core/utils';
+
+import './style.scss';
+
+
+export class AddonCompatibilityErrorBase extends React.Component {
+  static propTypes = {
+    i18n: PropTypes.object.isRequired,
+    lang: PropTypes.string.isRequired,
+    log: PropTypes.object,
+    maxVersion: PropTypes.string.isRequired,
+    minVersion: PropTypes.string.isRequired,
+    reason: PropTypes.string.isRequired,
+    userAgentInfo: PropTypes.object.isRequired,
+  }
+
+  static defaultProps = {
+    log: _log,
+  }
+
+  render() {
+    const {
+      i18n,
+      lang,
+      log,
+      maxVersion,
+      minVersion,
+      reason,
+      userAgentInfo,
```

Comment:
If `navigator.userAgent` is undefined the app init will still set it to `{ browser, os } = UAParser(undefined)`. That will be a lot of null values but it'll be set, as the start of every request sets the state of `api.userAgent` and `api.userAgentInfo`. So this should never be empty.

But I'll add the empty object anyway, doesn't hurt.